homeassistant/components/tankerkoenig/config_flow.py

```diff
@@ -144,6 +144,30 @@ async def async_step_select_station(
             options={CONF_SHOW_ON_MAP: True},
         )
 
+    async def async_step_reauth(self, data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Perform reauth confirm upon an API authentication error."""
+        if not user_input:
+            return self._show_form_reauth()
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert entry
+        user_input = {**entry.data, **user_input}
+        data = await async_get_nearby_stations(self.hass, user_input)
+        if not data.get("ok"):
+            return self._show_form_reauth(user_input, {CONF_API_KEY: "invalid_auth"})
+
+        self.hass.config_entries.async_update_entry(
+            entry, data=user_input, options=entry.options
```

Comment:
We only need to pass the updated data. If the options are the same we don't need to pass those.